File: Navigator.java
import java.lang.Math;

public class Navigator {

    Integer xPos;
    Integer yPos;
    Integer baseSpeed;
    Integer plugTime;
    boolean availability;

    public Navigator(Integer xPos, Integer yPos, Integer baseSpeed, 
                    Integer plugTime, boolean availability){
        this.xPos = xPos;
        this.yPos = yPos;
        this.baseSpeed = baseSpeed;
        this.plugTime = plugTime;
        this.availability = availability;
    }

    //goes to battery
    public Integer pickup(Battery batt){
        int xDelta = Math.abs(batt.xPos - this.xPos);
        int yDelta = Math.abs(batt.yPos - this.yPos);
        System.out.println("the x pos is: " + this.xPos);
        System.out.println("the x pos of the batt is: " + batt.xPos);
        // amount of time passed to pickup battery is the distance it takes to get there
        // (it travels along the x first, then the y), times the rate of travel
        Integer tDelta = (xDelta + yDelta) * this.baseSpeed;
        //moves the navigator to the battery module
        this.xPos = batt.xPos;
        this.yPos = batt.yPos;

        //prints the time it took for debugging
        System.out.println("the change in time is: " + tDelta);
        //returns the time that action took the system
        return tDelta;
    }

    public void reportStats(){
        System.out.println("REPORTING STATUS: " + "My current position is:  " + this.xPos + ", " + this.yPos);
    }

    //similar to pickup but for getting to the car with the new battery
    public Integer moveTo(Car c) {
        int xDelta = Math.abs(c.xPos - this.xPos);
        int yDelta = Math.abs(c.yPos - this.yPos);
        Integer tDelta = (xDelta + yDelta) * this.baseSpeed;
        this.xPos = c.xPos;
        this.yPos = c.yPos;

        //prints the time it took for debugging
        System.out.println("the change in time is: " + tDelta);
        //returns the time that action took the system
        return tDelta;
    }
}

File: Navigator.py
class Navigator:
    # Constructor
    def __init__(self, yPos, xPos, xSpeed, ySpeed, plugTime, available):
        self.xPos = xPos
        self.yPos = yPos
        self.xSpeed = xSpeed
        self.ySpeed = ySpeed
        self.plugTime = plugTime
        self.available = available

    def reportStats(self):
        print("REPORTING STATUS: " + "my current pos is " + str(self.xPos) + ", " + str(self.yPos))
    
    # 
    def moveTo(self, obj):
        xDelta = abs(obj.xPos - self.xPos)
        yDelta = abs(obj.yPos - self.yPos)
        # time passed to move to obj is the (distance along x + 
        # distance along y) / the rates of travel

        print("----DEBUGGING MOVETO------"+
              "\nObj pos:" + str(obj.xPos) + "," + str(obj.yPos) +
              "\nNav pos:" + str(self.xPos) + "," + str(self.yPos) +
              "\nxDelta = " + str(xDelta) +
              "\nyDelta = " + str(yDelta) +
              "\nxSpeed = " + str(self.xSpeed) +
              "\nySpeed = " + str(self.ySpeed))
        
        tDelta = (xDelta / self.xSpeed) + (yDelta / self.ySpeed)
        # moves the navigator to the object
        self.xPos = obj.xPos
        self.yPos = obj.yPos

        # print("moving there took " + str(tDelta) + " time")
        return tDelta

    def goHome(self):
        xDelta = abs(self.xPos)
        yDelta = abs(self.yPos)
        # time passed to move to obj is the (distance along x + 
        # distance along y) / the rates of travel
        tDelta = (xDelta / self.xSpeed) + (yDelta / self.ySpeed)
        # moves the navigator to the object
        self.xPos = 0
        self.yPos = 0

        # print("moving there took " + str(tDelta) + " time")
        return tDelta


    def useArm(self):
        tDelta = self.plugTime
        return tDelta